Error: unable to build kubernetes objects from release manifest: error validating ""

Please, answer some short questions which should help us to understand your problem / question better?

- **Which image of the operator are you using?** registry.opensource.zalan.do/acid/postgres-operator:v1.5.0
- **Where do you run it - cloud or metal? Kubernetes or OpenShift?** Bare Metal Openshift
- **Are you running Postgres Operator in production?** no
- **Type of issue?** Bug report and/or question

Trying to install the operator in my Openshift environment (4.5.0-0.okd-2020-10-29-044318) I am getting the following error message:

```
git clone https://github.com/zalando/postgres-operator.git
cd postgres-operator.git
helm install postgres-operator ./charts/postgres-operator -f ./charts/postgres-operator/values-crd.yaml -n "${_namespace}" --create-namespace

coalesce.go:199: warning: destination for cluster_labels is a table. Ignoring non-table value application:spilo
manifest_sorter.go:192: info: skipping unknown hook: "crd-install"
manifest_sorter.go:192: info: skipping unknown hook: "crd-install"
manifest_sorter.go:192: info: skipping unknown hook: "crd-install"
Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: [ValidationError(OperatorConfiguration.configuration.aws_or_gcp): unknown field "enable_ebs_gp3_migration" in do.zalan.acid.v1.OperatorConfiguration.configuration.aws_or_gcp, ValidationError(OperatorConfiguration.configuration): unknown field "enable_pgversion_env_var" in do.zalan.acid.v1.OperatorConfiguration.configuration]

```
Greeping through `charts/postgres-operator/crds/operatorconfigurations.yaml` shows me no match for `enable_pgversion_env_var`.

--- a/charts/postgres-operator/crds/operatorconfigurations.yaml
+++ b/charts/postgres-operator/crds/operatorconfigurations.yaml
@@ -69,6 +69,8 @@ spec:
                 type: boolean
               enable_lazy_spilo_upgrade:
                 type: boolean
+              enable_pgversion_env_var:
+                type: boolean
               enable_shm_volume:
                 type: boolean
               etcd_host:

--- a/charts/postgres-operator/values-crd.yaml
+++ b/charts/postgres-operator/values-crd.yaml
@@ -22,7 +22,7 @@ configGeneral:
   # update only the statefulsets without immediately doing the rolling update
   enable_lazy_spilo_upgrade: false
   # set the PGVERSION env var instead of providing the version via postgresql.bin_dir in SPILO_CONFIGURATION
-  enable_pgversion_env_var: "false"
+  enable_pgversion_env_var: false
   # start any new database pod without limitations on shm memory
   enable_shm_volume: true
   # etcd connection string for Patroni. Empty uses K8s-native DCS.
@@ -270,7 +270,7 @@ configTeamsApi:
   # operator watches for PostgresTeam CRs to assign additional teams and members to clusters
   enable_postgres_team_crd: false
   # toogle to create additional superuser teams from PostgresTeam CRs
-  # enable_postgres_team_crd_superusers: "false"
+  # enable_postgres_team_crd_superusers: false
 
   # toggle to grant superuser to team members created from the Teams API
   enable_team_superuser: false

--- a/charts/postgres-operator/values.yaml
+++ b/charts/postgres-operator/values.yaml
@@ -217,7 +217,7 @@ configAwsOrGcp:
   # enable automatic migration on AWS from gp2 to gp3 volumes
   enable_ebs_gp3_migration: "false"
   # defines maximum volume size in GB until which auto migration happens
-  # enable_ebs_gp3_migration_max_size: 1000
+  # enable_ebs_gp3_migration_max_size: "1000"
 
   # GCP credentials for setting the GOOGLE_APPLICATION_CREDNETIALS environment variable
   # gcp_credentials: ""

--- a/manifests/configmap.yaml
+++ b/manifests/configmap.yaml
@@ -37,7 +37,7 @@ data:
   # enable_crd_validation: "true"
   # enable_database_access: "true"
   enable_ebs_gp3_migration: "false"
-  # enable_ebs_gp3_migration_max_size: 1000
+  # enable_ebs_gp3_migration_max_size: "1000"
   # enable_init_containers: "true"
   # enable_lazy_spilo_upgrade: "false"
   enable_master_load_balancer: "false"

--- a/manifests/configmap.yaml
+++ b/manifests/configmap.yaml
@@ -15,7 +15,7 @@ data:
   # connection_pooler_default_cpu_request: "500m"
   # connection_pooler_default_memory_limit: 100Mi
   # connection_pooler_default_memory_request: 100Mi
-  connection_pooler_image: "registry.opensource.zalan.do/acid/pgbouncer:master-11"
+  connection_pooler_image: "registry.opensource.zalan.do/acid/pgbouncer:master-12"
   # connection_pooler_max_db_connections: 60
   # connection_pooler_mode: "transaction"
   # connection_pooler_number_of_instances: 2
@@ -41,6 +41,7 @@ data:
   # enable_init_containers: "true"
   # enable_lazy_spilo_upgrade: "false"
   enable_master_load_balancer: "false"
+  # enable_pgversion_env_var: "false"
   # enable_pod_antiaffinity: "false"
   # enable_pod_disruption_budget: "true"
   # enable_postgres_team_crd: "false"

--- a/manifests/minimal-fake-pooler-deployment.yaml
+++ b/manifests/minimal-fake-pooler-deployment.yaml
@@ -23,7 +23,7 @@ spec:
       serviceAccountName: postgres-operator
       containers:
       - name: postgres-operator
-        image: registry.opensource.zalan.do/acid/pgbouncer:master-11
+        image: registry.opensource.zalan.do/acid/pgbouncer:master-12
         imagePullPolicy: IfNotPresent
         resources:
           requests:

--- a/manifests/postgresql-operator-default-configuration.yaml
+++ b/manifests/postgresql-operator-default-configuration.yaml
@@ -6,6 +6,7 @@ configuration:
   docker_image: registry.opensource.zalan.do/acid/spilo-12:1.6-p3
   # enable_crd_validation: true
   # enable_lazy_spilo_upgrade: false
+  # enable_pgversion_env_var: false
   # enable_shm_volume: true
   etcd_host: ""
   # kubernetes_use_configmaps: false

--- a/pkg/util/constants/pooler.go
+++ b/pkg/util/constants/pooler.go
@@ -14,5 +14,5 @@ const (
 	ConnectionPoolerContainer            = 0
 	ConnectionPoolerMaxDBConnections     = 60
 	ConnectionPoolerMaxClientConnections = 10000
-	ConnectionPoolerMinInstances         = 2
+	ConnectionPoolerMinInstances         = 1
 )

